### PR TITLE
Introduce Google test and Google mock infra

### PR DESCRIPTION
**Pull request by githublaohu**
Introduction of test library Google test and google mock to provide unit and integration test functions

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+include_directories(${CMAKE_SOURCE_DIR}/include)
+include_directories(${SUB_DIRS})
+
+set(EXECUTABLE_OUTPUT_PATH ${CMAKE_SOURCE_DIR}/test/bin)
+
+include_directories(${CMAKE_SOURCE_DIR}/include)
+include_directories(${Boost_INCLUDE_DIRS})
+
+
+set(Gtest_INCLUDE_DIR    ${CMAKE_SOURCE_DIR}/bin/include/gtest)
```
Comment (jovan-wong) [inline test/CMakeLists.txt]: `set(Gtest_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/bin/include/gtest)`
Just use 1 space between 2 args.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    	else()
+        	target_link_libraries (${basename}  rocketmq_static ${deplibs}
+        	${Boost_LIBRARIES} ${LIBEVENT_LIBRARIES} ${JSONCPP_LIBRARIES} ${Gtest_LIBRARIES})
+        endif()
+    else()
+        #if (BUILD_ROCKETMQ_SHARED)
+            target_link_libraries (${basename}  rocketmq_shared ${deplibs})
+            target_link_libraries (${basename}  rocketmq_shared ${Gtest_LIBRARIES})
+            target_link_libraries (${basename}  rocketmq_shared ${Gmock_LIBRARIES})
+        #else()
```
Comment (jovan-wong) [inline test/CMakeLists.txt]: If these codes are useless, could you plz remove them?

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+set(Gtest_INCLUDE_DIR    ${CMAKE_SOURCE_DIR}/bin/include/gtest)
+set(Gmock_INCLUDE_DIR    ${CMAKE_SOURCE_DIR}/bin/include/gmock)
+
+include_directories(${Gtest_INCLUDE_DIR})
+include_directories(${Gtest_INCLUDE_DIR}/internal)
+include_directories(${Gmock_INCLUDE_DIR})
+include_directories(${Gmock_INCLUDE_DIR}/internal)
+
+set(Gtest_LIBRARY_DIRS   ${CMAKE_SOURCE_DIR}/bin/lib)
```
Comment (jovan-wong) [inline test/CMakeLists.txt]: also here

File: build.sh [hunk 4]
```diff
@@ -121,7 +124,7 @@ Prepare()
     else
         mkdir -p ${build_dir}
     fi
-	
+
```
[the remark this answers is not in the document]
Comment (githublaohu) [reply build.sh]: Alerady delete

**Pull request merged**